gmso/utils/misc.py

```diff
@@ -47,3 +47,61 @@ def ensure_valid_dimensions(quantity_1: u.unyt_quantity,
             quantity_2.units,
             quantity_2.units.dimensions
         )
+
+
+def validate_type(iterator, type_):
+    """Validate all the elements of the iterable are of a particular type"""
+    for item in iterator:
+        if not isinstance(item, type_):
+            raise TypeError(
+                f"Expected {item} to be of type {type_.__name__} but got"
+                f" {type(item).__name__} instead."
+            )
+
+
+def mask_with(iterable, window_size=1, mask='*'):
+    """Mask an iterable with the `mask` in a circular sliding window of size `window_size`
+
+    This method masks an iterable elements with a mask object in a circular sliding window
+
+    Parameters
+    ----------
+    iterable: Iterable
+        The iterable to mask with
+    window_size: int, default=1
+        The window size for the mask to be applied
+    mask: Any, default='*'
+        The mask to apply
+    Examples
+    --------
+        >>> from gmso.utils.misc import mask_with
+        >>> list(mask_with(['Ar', 'Ar'], 1))
+        [['*', 'Ar'], ['Ar', '*']]
+        >>> for masked_list in mask_with(['Ar', 'Xe', 'Xm', 'CH'], 2, mask='_'):
+        ...     print('~'.join(masked_list))
+        _~_~Xm~CH
+        Ar~_~_~CH
+        Ar~Xe~_~_
+        _~Xe~Xm~_
```

Comment:
I think this work for our case (`_~Xe~Xm~_`), but there is one two cases that this does not cover (`_~Xe~_~CH` and `Ar~_~Xm~_`). Just a note for now since I am not sure we will encounter these cases to put in the work.